sangyeob/app.js

```diff
@@ -0,0 +1,45 @@
+const http = require('http');
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+
+const dotenv = require('dotenv');
+dotenv.config();
+
+const { DataSource } = require('typeorm');
+
+const myDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE,
+});
+
+myDataSource.initialize().then(() => {
+    console.log('Data Source has been initialized!');
+});
```

Comment:
- myDataSource 보다는 `appDataSource` , `mysqlDataSource` 혹은 간결하게 `database`로 변수명을 변경해주세요. “my”라는 용어는 예제 코드에서는 사용할 수 있으나 실제 코드에서는 피해주세요.
- DataSource.initialize() 함수가 실행되고 나면, then()과 catch()를 사용해서 에러를 핸들링해줄 수 있습니다. 이렇게 하면, initialize() 함수가 실행되어서 database와 실제로 connection을 생성할 때, 성공했는지 실패했는지 여부를 확인할 수 있습니다.
- 추후 에러 핸들링을 담을 때의 가독성 및 확장성 보장을 위해 ".then"을 (typeorm datasource 공식문서에서 제안하는 것 처럼) 한칸 아래로 띄는 것을 권장드립니다.
- 참고 링크: [https://typeorm.biunav.com/en/data-source.html#what-is-datasource](https://typeorm.biunav.com/en/data-source.html#what-is-datasource)

```suggestion
const appDataSource = new DataSource({
    type: process.env.TYPEORM_CONNECTION,
    host: process.env.TYPEORM_HOST,
    port: process.env.TYPEORM_PORT,
    username: process.env.TYPEORM_USERNAME,
    password: process.env.TYPEORM_PASSWORD,
    database: process.env.TYPEORM_DATABASE,
});

appDataSource.initialize().then(() => {
    console.log('Data Source has been initialized!');
}); // .catch() 로 에러핸들링 코드 추가!
```